libhoney/transmission.py

```diff
@@ -92,23 +97,29 @@ def _send(self, ev):
             "body": resp.text,
             "error": "",
         }
-        try:
-            if self.block_on_response:
-                self.responses.put(response)
-            else:
+        if self.block_on_response:
+            self.responses.put(response)
+        else:
+            try:
                 self.responses.put_nowait(response)
-        except queue.Full:
-            pass
+            except queue.Full:
+                pass
 
     def close(self):
         '''call close to send all in-flight requests and shut down the
            senders nicely'''
         for i in range(self.max_concurrent_batches):
-            self.pending.put(None)
+            try:
+                self.pending.put(None, true, 10)
+            except queue.Full:
+                pass
```

Comment:
if we give up here and continue, the `t.join`s below will hang.  looks like `Thread#join` can take a timeout as well, though.